experimental/gracefulblobmissing/patches.py

```diff
@@ -238,9 +260,37 @@ def patched_loadBlob_relstorage(self, cursor, oid, serial):
         lock.close()
 
 
-def create_empty_blob(filename):
-    mkdir("-p", '/'.join(filename.split('/')[:-1]))
-    fo = open(filename, 'w')
-    fo.write("File created by experimental.gracefulblobmissing.")
-    fo.close()
-    logger.info("Created missing blob-file %s" % filename)
+def guess_extension(blob=None):
+    ''' Try to guess the type of the blob out of thin air
+    '''
+    request = getRequest()
+    if not request:
+        return 'txt'
+    try:
+        published = repr(request.PUBLISHED.im_class)
```

Comment:
This fails for me. I have a `SimpleviewClass` which has no `im_func`. I think this is a Python 2-3 difference. (I am on Python 3.)

Reply:
Most probably `im_class` should become `__class__`. I am not sure why I wrote it like that :)

Reply:
I have removed the entire function and always do a PNG now, as that works for both images and files.
Is this okay for you?